### PR TITLE
Pass resource client to helm/template controllers instead of dynamic client

### PR DESCRIPTION
**Pull request by chancez**
Dynamic client is more than we need, and the resourceClient can be
instantiated with everything it needs in buildCRWatcher.

### CODE REVIEW

File: helmctlr/helm.go [hunk 1]
```diff
@@ -53,15 +53,15 @@ func NewController(chartDir, ns, rn, host string, wait bool, waitto int64, logge
...
-		kubeClient:  kubeClient,
-		logger:      logger,
+		Helm:           helm.NewClient(helm.Host(host)),
+		ChartDir:       chartDir,
+		Namespace:      ns,
+		ReleaseName:    rn,
+		Wait:           wait,
+		WaitTimeout:    waitto,
+		resourceClient: resourceClient,
+		kubeClient:     kubeClient,
```
Comment (alecmerdler) [inline helmctlr/helm.go:63]: Remove `kubeClient`, it is not used or needed.

**Review by ecordell — APPROVED** [review]
LGTM

**Review by alecmerdler — DISMISSED** [review]
Remove unneeded `kubeClient`.

File: helmctlr/helm.go (hunk 1 above)
Comment (ecordell) [reply helmctlr/helm.go:63]: we'll need kubeclient for the work Chance is about to do to write to configmaps, unless I misunderstood

Comment (alecmerdler) [reply helmctlr/helm.go:63]: Then we should pass the `ResourceInterface` for `ConfigMaps`, not the full k8s client.

Comment (chancez) [reply helmctlr/helm.go:63]: The `fake` client works on this interface, so I'd prefer to keep it as it is.

**Pull request merged**